test/unit/shared/access_permissions_test_base.rb

```diff
@@ -21,10 +21,15 @@ def check_routes(app_routes, skipped_actions)
           controller = "#{r.defaults[:controller]}_controller".classify.constantize
           filters    = controller.send(:_process_action_callbacks)
 
-          # Pass if the controller deliberately only permit admins (e.g. SettingsController)
-          if filters.select { |f| f.filter == :require_admin }.empty?
-            assert_not_equal [], Foreman::AccessControl.permissions.select { |p| p.actions.include? path }
-          end
+          permission_required = true
+
+          # Pass if the controller only permit admins (e.g. SettingsController)
+          permission_required = false if filters.select { |f| f.filter == :require_admin }.any?
```

Comment:
you can use `filters.detect{}` instead of `filters.select{}.any?`

Reply:
sorry, `filters.any?{}` instead of `filters.select{}.any?` so that the loop stops once that filter is found.